File: Praktikum 6. Array/Soal 4/PRAK604-2310817310001-AvantioFierzaPatria.py
def decode_message(code, message):
    decoded_message = ""
    star_count = 0
    hash_count = 0

    for i in range(len(code)):
        if code[i] == message[i]:
            decoded_message += '*'
            star_count += 1
        else:
            decoded_message += '#'
            hash_count += 1

    result_message = "Rangkaian Karakter: " + decoded_message
    result_star_count = "Jumlah Bintang: " + str(star_count)
    result_hash_count = "Jumlah Pagar: " + str(hash_count)

    if star_count >= hash_count:
        result_status = "Pesan Asli"
    else:
        result_status = "Pesan Palsu"

    return result_message, result_star_count, result_hash_count, result_status


# Meminta input dari pengguna
code = input("Masukkan kode Shikamaru: ")
message = input("Masukkan pesan yang diterima: ")

# Memanggil fungsi dan menampilkan hasil
result_message, result_star_count, result_hash_count, result_status = decode_message(
    code, message)

print(result_message)
print(result_star_count)
print(result_hash_count)
print(result_status)

File: Praktikum 6. Array/Soal 4/PRAK604-2310817310001-AvantioFierzaPatria.c
#include <stdio.h>
#include <string.h>

void decodeMessage(char code[], char receivedMessage[], char decodedMessage[], int *starCount, int *hashCount)
{
    int length = strlen(code);
    *starCount = 0;
    *hashCount = 0;

    for (int i = 0; i < length; i++)
    {
        if (code[i] == receivedMessage[i])
        {
            decodedMessage[i] = '*';
            (*starCount)++;
        }
        else
        {
            decodedMessage[i] = '#';
            (*hashCount)++;
        }
    }

    decodedMessage[length] = '\0';
}

int main()
{
    char code[100], receivedMessage[100], decodedMessage[100];
    int starCount, hashCount;

    printf("Masukkan kode Shikamaru: ");
    scanf("%s", code);
    printf("Masukkan pesan yang diterima: ");
    scanf("%s", receivedMessage);

    decodeMessage(code, receivedMessage, decodedMessage, &starCount, &hashCount);

    printf("%s\n", decodedMessage);
    printf("Jumlah tanda bintang: %d\n", starCount);
    printf("Jumlah tanda pagar: %d\n", hashCount);

    if (starCount >= hashCount)
    {
        printf("Pesan asli\n");
    }
    else
    {
        printf("Pesan palsu\n");
    }

    return 0;
}
